# dnaseq

*under active development*

Snakemake based workflow for somatic and germline variant calling.



### TODO

- [x] Somatic SNP/small indels calling 
- [ ] Germline SNP/small indels calling
- [ ] structural variant calling
- [ ] cnv calling



### Credits

The pipeline is based upon the [dna-seq-gatk-variant-calling](https://github.com/snakemake-workflows/dna-seq-gatk-variant-calling) best practices workflow. It also reuses [test data](https://github.com/snakemake-workflows/ngs-test-data) from this repository for travis CI testing.
